src/main/java/com/google/devtools/build/lib/rules/cpp/CppLinkAction.java

```diff
@@ -538,9 +540,18 @@ private LazyData doCostlyEstimation() {
       }
 
       // TODO(https://github.com/bazelbuild/bazel/issues/17368): Use inputBytes in the computations.
-
-      double ramUsageMbs = 100.0 + 2.4 * (inputsBytes / 1024.0 / 1024.0);
+      double coefficient = 2.4;
+      if (executionInfo.containsKey("coefficient")){
+        coefficient = Double.valueOf(executionInfo.get("coefficient"));
```

Comment:
the properties here are part of the `cpp_link` exec prop group, so in the mongo server build you set them it via `cpp_link.coefficient` in the exec_properties.